addons/product/models/product.py

```diff
@@ -754,3 +753,10 @@ def get_import_templates(self):
             'label': _('Import Template for Vendor Pricelists'),
             'template': '/product/static/xls/product_supplierinfo.xls'
         }]
+
+    @api.model_create_single
+    def create(self, vals):
+        if 'product_uom' not in vals:
+            vals['product_uom'] = self.env['product.template'].browse(vals['product_tmpl_id']).uom_po_id.id
```

Comment:
Thinking again, maybe this implementation is prone to errors since neither product_tmpl_id or product_id are required fields. So that's what's happening with the tests errors: supplierinfos are created with no product detailed and assigned later.

So maybe we need a different aproach to ensure a UoM whenever the product or the template is loaded and using the new v13 framework behavior. So, the field could be defined like:

```python
    product_uom = fields.Many2one(
        'uom.uom', 'Unit of Measure',
        compute='_compute_product_uom',
        store=True, readonly=False,
        help="This comes from the product form.")
```

And the compute would be something of the sort:

```python
api.depends('product_tmpl_id')
def _compute_product_uom:
    for rec in self:
        if rec.product_tmpl_id:
            rec.product_uom = rec.product_tmpl_id.uom_po_id
```

So the value will default to the one in product_tmpl_id as before, but can be set independently.